### PR TITLE
Remove fixed screen orientation

### LINKED ISSUE #21084: Consider dropping `android:screenOrientation` parameter
### Expected behavior
No activity in `AndroidManifest.xml` files has `android:screenOrientation`

### Actual behavior
Some activities have defined either `landscape` or `portrait` modes. Citing Android Lint documentation:

> Should not restrict activity to fixed orientation. This may not be suitable for different form factors, causing the app to be letterboxed.

### Steps to reproduce the behavior
Remove `DiscouragedApi` issues from baseline and run a lint task (after #21082 is merged).

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #21084 

Previously we had several screens fixed into either portrait or landscape. This PR addresses this by removing all fixed orientation. The following screens are affected:

* **ReaderVideoViewerActivity** - I could never get this activity to appear. I created a blog post with a video but when I viewed that post in the reader and tapped a video to make it full-screen that activity didn't appear. Likewise, videos in other reader posts didn't result in this activity appearing. So I opted to force it to appear by adding `ReaderActivityLauncher.showReaderVideoViewer(requireActivity(), "https://youtu.be/eWRn80UOsqw?si=78nqh1LqykHXqHtV")` to the end of `MySite.onViewCreated` - and it crashed with an NPE! I fixed that and it worked just fine in both portrait and landscape.
* **Blaze screens** - these had issues with being rotated due to state not being saved & restored, but this has been addressed
* **SupportWebViewActivity** - I was unable to test this (in my case, "Contact support" doesn't work in debug builds), so I'd appreciate the reviewer testing this one
* **JetpackMigrationActivity** - this is a hard one to test, so I recommend adding `showJetpackMigrationDeleteWP()` at the end of `MySite.onViewCreated` to force it to appear

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/blaze/blazepromote/BlazePromoteParentActivity.kt [hunk 1]
```diff
@@ -40,6 +63,8 @@ class BlazePromoteParentActivity : LocaleAwareActivity() {
                     supportFragmentManager.beginTransaction()
                         .replace(R.id.container, BlazePromoteWebViewFragment.newInstance())
                         .commitNow()
+                    // without this, rotating the screen will show the overlay again
+                    shouldShowOverlay = false
```
Comment (dcalhoun) [inline WordPress/src/main/java/org/wordpress/android/ui/blaze/blazepromote/BlazePromoteParentActivity.kt:67]: Noting that this appears to also dismiss the overlay if the device is rotated while it is displayed. Likely not a significant issue, but could be disruptive is someone accidentally rotates their device while reading the overlay. Unsure if this is worth spending time to address.

**Review by dcalhoun — COMMENTED** [review]
The changes look sound to me. On my Samsung device, the YouTube video launched in Chrome when I used the testing instructions. Is that expected? 


<details><summary>YouTube video screen recording</summary>
<p>

https://github.com/user-attachments/assets/b226f6ac-2687-4f31-a846-850a4f6f2f7d

</p>
</details>

**Comment by nbradbury** [comment]
> On my Samsung device, the YouTube video launched in Chrome when I used the testing instructions. Is that expected?

That's unexpected but nothing that would've been the result of this PR. Note that on my Samsung, it plays in the app.

**Review by dcalhoun — APPROVED** [review]

**Pull request merged**